services/services.go

```diff
@@ -165,4 +165,11 @@ type IUserService interface {
 	GenerateResetToken(*models.User) (*models.User, error)
 	FlushCache()
 	FlushUserCache(string)
+	WebAuthnBeginRegistration(user *models.User, credentialName string) (credentialCreationOptions interface{}, sessionData interface{}, err error)
```

Comment:
Please use proper typing instead of `interface{}` (also for the methods below). Also, I rather see these methods inside the `routes` package (right next to the actual endpoints), because the are very specific for the WebAuthn flow and only remotely related to functionality of the `User`.